Remove `friend class` from the library wording (#6427)
cplusplus/draft: source/iostreams.tex

\pnum
\begin{note}

> \tcode{directory_iterator} is allowed to cache

Reviewer: Are we allowed to say "allowed"?
Should this be turned into a "Recommended practice" note and say something like "Implementations of `directory_iterator` are encouraged to populate attributes of the `directory_entry` object upon construction, without calling `refresh()`"?
OTOH does this note actually mean anything at all? Implementations are _always_ encouraged to omit "unneeded" code; we don't have to tell them that, do we?

Author: Changed to use "can". (So did #6758.)

\pnum
\begin{note}

> \tcode{directory_iterator} can cache

Reviewer 2: Please squash this into the earlier commit that is fixed by this change.

Then, force-push.

Author: Done.

\pnum
\begin{note}
\tcode{directory_iterator} can cache
already available attribute values
directly into a \tcode{directory_entry} object

> without the cost of an unneeded call to \tcode{refresh()}.

Reviewer: ```suggestion
without the cost of a call to \tcode{refresh()}.
```

Author: Moved the change to #7245.

\tcode{directory_iterator} can cache
already available attribute values
directly into a \tcode{directory_entry} object

> without the cost of a call to \tcode{refresh()}.

Reviewer 2: This has nothing to do with the topic of this pull request, which is "friend class" removal.

Please move this into a separate pull request and format the change description appropriately.

Author: Done. Moved the change to #7245.
